src/bootstrap/check.rs

```diff
@@ -424,6 +424,40 @@ fn path_for_cargo(builder: &Builder, compiler: Compiler) -> OsString {
     env::join_paths(iter::once(path).chain(env::split_paths(&old_path))).expect("")
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct RustdocJS {
+    pub host: Interned<String>,
+}
+
+impl Step for RustdocJS {
+    type Output = PathBuf;
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun) -> ShouldRun {
+        run.path("node")
+    }
+
+    fn make_run(run: RunConfig) {
+        run.builder.ensure(RustdocJS {
+            host: run.host,
+        });
+    }
+
+    fn run(self, _: &Builder) {
```

Comment:
* The `_` probably needs to be `builder` for line 457 to work.
* This function needs to return `Self::Output` anyway, for the trait to match. Since `Self::Output` is `PathBuf`, that's what this needs to return. However, all of the other `Step` impls in this file return `()`, so that's probably what the `Output` needs to be here as well.

Reply:
Certainly, I stopped trying to make it work and certainly forgot to commit the version which compiles. The problem is more global anyway. But thanks for taking a look in here. ;)